### ISSUE TITLE
TLS weird hostname translation to IP

### ISSUE THREAD

**Issue by adrianlzt** [skydive-project/skydive#2249]
When configuring Skydive to use TLS I have found an strange behaviour.

The config file looks like:
```
tls:
  client_cert: /path/certs/client/client.pem
  client_key: /path/certs/client/client.key

  server_cert: /path/server.pem
  server_key: /path/server.key
  ca_cert: /path/ca.pem

analyzers:
  - foo.domain.com:8082

...
```

(foo.domain.com resolves to 127.0.0.1)

If I start skydive analyzer it return an error, complainig because the invalid cert:
```
2020-07-09T11:45:05.079+0200    INFO    analyzer/analyzer.go:55 glob..func1     archer: Skydive Analyzer started !
2020-07-09T11:45:05.079+0200    INFO    hub/replication_endpoint.go:141 (*ReplicatorPeer).connect       archer: Connecting to peer: wss://127.0.0.1:8082/ws/replication
2020-07-09T11:45:05.080+0200    INFO    websocket/client.go:528 (*Client).Connect       archer: Connecting to wss://127.0.0.1:8082/ws/replication
2020-07-09T11:45:05.144+0200    ERROR   websocket/client.go:585 (*Client).Start.func1   archer: Unable to create a WebSocket connection wss://127.0.0.1:8082/ws/replication : x509: cannot validate certificate for 127.0.0.1 because it doesn't contain any IP SANs
```

Looking at the code I have found the function [AddressFromString](https://github.com/skydive-project/skydive/blob/d455b0cd07fd317fd0011305d8afd458f8365184/graffiti/service/address.go#L37) which is converting the hostname to an IP address.

Why is that needed?

Removing that translation from host to IP it works correctly.

**Comment by adrianlzt** [comment]
The _patch_:
```
diff --git a/graffiti/service/address.go b/graffiti/service/address.go
index 882714be..81fd774c 100644
--- a/graffiti/service/address.go
+++ b/graffiti/service/address.go
@@ -61,10 +61,10 @@ func AddressFromString(addressPort string) (Address, error) {
        }

        // just take the first address returned
-       addr := normalizeIPForURL(ips[0])
+       _ = normalizeIPForURL(ips[0])

        return Address{
-               Addr: addr,
+               Addr: host, // NOCOMMIT prueba a ver si usa el hostname
                Port: portNum,
        }, nil
 }
```

**Comment by safchain** [comment]
You could try to send a PR remove the call of this function so that the CI will test the fix. The scale job tests TLS

**Comment by adrianlzt** [comment]
#2257

**Comment by adrianlzt** [comment]
Closed in #2257
